teachers_digital_platform/js/scroll.test.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const scroll = require( './scroll.js' );
+
+const HTML_SNIPPET = `
+  <a class="one" href="#one" data-scroll>one</a>
+  <a class="two" href="#two" data-scroll>two</a>
+  <a class="three" href="#three" data-scroll>three</a>
+  <div id="one"></div>
+  <div id="two"></div>
+  <div id="three"></div>
+`;
+
+// Helper function to simulate events
+function simulateEvent( eventType, target, eventOption ) {
+  const event = document.createEvent( 'Event' );
+  if ( eventOption && eventOption.keyCode ) {
+    event.keyCode = eventOption.keyCode;
+  }
+  event.initEvent( eventType, true, true );
+  return target.dispatchEvent( event );
+}
+
+/* eslint-disable max-lines-per-function, no-undefined */
+describe( 'Scroll', () => {
+
+  beforeEach( () => {
+    // Load HTML fixture
+    document.body.innerHTML = HTML_SNIPPET;
+    global.scroll = jest.fn();
+    scroll.init();
+  } );
+
+  afterEach( () => {
+    document.body.innerHTML = '';
+  } );
+
+  it( 'should update the URL when clicked', () => {
+    const spy = jest.spyOn( global.history, 'pushState' );
```

Comment:
Jest spies are a little unusual in that they [call the method being spied on](https://jestjs.io/docs/en/jest-object.html#jestspyonobject-methodname) instead of overriding it.